restapi/stories.go

```diff
@@ -34,8 +34,65 @@ func init() {
 
 func (api api) ReturnAllStories(w http.ResponseWriter, r *http.Request) render.Renderer {
 	var stories []models.Story
+	var storiesByTags []models.Tag
 
-	err := api.database.Where("is_visible = true").Find(&stories).Error
+	getVisibility := r.URL.Query()["visibility"]
+	visibility := true
+	if getVisibility != nil {
+		vb, err := strconv.ParseBool(getVisibility[0])
+		if err != nil {
+			return rest.ErrInternal(api.logger, err)
+		}
+		visibility = vb
+	}
+
+	err := r.ParseForm()
+	if err != nil {
+		return rest.ErrInternal(api.logger, err)
+	}
+	sort := r.Form["sort"]
+	order := r.Form["order"]
+	tags := r.Form["tags"]
+
+	sortString := ""
+	for i := 0; i < len(sort); i++ {
+		//special case for author name because our db table doesnt have "name" in one column
+		if sort[i] == "author_name" {
+			sortString += "author_first_name" + " " + order[i] + ", " + "author_last_name" + " " + order[i]
+		} else {
+			sortString += sort[i] + " " + order[i]
+		}
+
+		if i != len(sort)-1 {
+			sortString += ", "
+		}
+	}
+
+	gormStories := api.database.Table("stories")
+	gormTags := api.database.Table("tags")
```

Comment:
We probably dont need to query Tags table in return stories endpoint. If you look at my PR, we now return tags as part of a new struct called Taggedstory. So in terms of sorting and querying by tag, we can use that. We can discuss this more if you are confused about this.

Reply:
The thing is we don't want to manually pull all stories with tags then query them when they're in an array. We should call your function at the very end and instead use sql queries to do the filtering and sorting (easier logic wise and faster)